pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -148,9 +180,54 @@ public void build()
       StarTreeIndexMapUtils
           .storeToFile(indexMaps, new File(_segmentDirectory, StarTreeV2Constants.INDEX_MAP_FILE_NAME));
       FileUtils.forceDelete(starTreeIndexDir);
+      _existingStarTrees.cleanOutputDirectory();
     }
 
-    LOGGER.info("Finished building {} star-trees in {}ms", numStarTrees, System.currentTimeMillis() - startTime);
+    LOGGER.info("Newly built start-trees {}.\n Reused star-trees {}.", numStarTrees - reusedStarTrees, reusedStarTrees);
+    LOGGER.info("Finished building star-trees in {}ms", System.currentTimeMillis() - startTime);
```

Comment:
(minor) Combine them into one single log line
`Finished building {} star-trees ({} reused) in {}ms`

Reply:
done